apps/sr/tid1500writer.cxx

```diff
@@ -105,15 +98,21 @@ int main(int argc, char** argv){
       cout << "Loading " << dicomFilePath << endl;
       CHECK_COND(ff.loadFile(dicomFilePath.c_str()));
 
-      DcmDataset imageLibDataset;
-      DcmTag tagsToCopy[] = {DCM_SOPClassUID,DCM_SOPInstanceUID,DCM_Modality,DCM_StudyDate,DCM_Columns,DCM_Rows,DCM_PixelSpacing,DCM_BodyPartExamined,DCM_ImageOrientationPatient,DCM_ImagePositionPatient};
-      for(int t=0;t<10;t++){
-        copyElement(tagsToCopy[t], ff.getDataset(), &imageLibDataset);
+
+      if(i==0){
+        DcmDataset imageLibGroupDataset;
+        DcmTag commonTagsToCopy[] =   {DCM_SOPClassUID,DCM_Modality,DCM_StudyDate,DCM_Columns,DCM_Rows,DCM_PixelSpacing,DCM_BodyPartExamined,DCM_ImageOrientationPatient};
+        for(int t=0;t<STATIC_ARRAY_SIZE(commonTagsToCopy);t++){
+          ff.getDataset()->findAndInsertCopyOfElement(commonTagsToCopy[t],&imageLibGroupDataset);
+        }
+        CHECK_COND(report.getImageLibrary().addImageEntryDescriptors(imageLibGroupDataset));
       }
 
-      OFCHECK(report.getImageLibrary().addImageGroup().good());
-      OFCHECK(report.getImageLibrary().addImageEntry(imageLibDataset).good());
-      OFCHECK(report.getImageLibrary().addImageEntryDescriptors(imageLibDataset).good());
+      DcmDataset imageEntryDataset;
+      DcmTag imageTagsToCopy[] = {DCM_Modality,DCM_SOPClassUID,DCM_SOPInstanceUID,DCM_ImagePositionPatient};
```

Comment:
Yes, this will also work. Navigating through the tree should be easy since all content items that belong to a template row are annotated by "TID xxxx - Row n". You can use gotoAnnotatedNode() and gotoNextAnnotatedNode() for this purpose. See test case "dcmsr_TID1001_ObservationContext". Of course, XPath expressions would also be nice... :)